### DISCUSSION TITLE
GridStat trouble reading probabilistic grib field

### DISCUSSION THREAD

**Discussion by lindsayrblank** [dtcenter/METplus#2107] [File I/O] [answered]
Hi MET team,

I am trying to calculate some probability statistics using the grid_stat tool. I have defined my forecast dictionary as such:

```
fcst = {

         name = "TMP";
         level = "Z2";
         prob = TRUE;
         cat_thresh = [ ==0.1 ];

     field = [
         { GRIB2_ipdtmpl_index=[ 19, 21 ]; GRIB2_ipdtmpl_val=[ 25815, 26815 ]; }
     ];
}
```

When I run grid_stat, it produces the following error:
```
WARNING: MetGrib2DataFile::data_plane() -> No matching record found for 'TMP/Z2'
WARNING: read_data_plane() -> TMP/Z2 not found in file: model1.t00z.pgrb2.0p125.f012
ERROR  : process_scores() -> no requested data found!  Exiting...
```

However, if I remove `prob = TRUE; cat_thresh = [ ==0.1 ];` it *does* find the data. Any idea what could be happening? I've attached the config file. I will send the model data over via the FTP.

Thanks!

[GridStatConfig_prob_test.txt](https://github.com/dtcenter/METplus/files/11163590/GridStatConfig_prob_test.txt)

**Comment by CPKalb** [comment, marked as answer]
Hi Lindsay:

Getting the format right for probabilities can be tough.   The True/False is for netCDF files, whereas grib/grib2 files need a dictionary specified in the prob variable.  I'm guessing you need to change the format of your input data to look something like:

fcst = {

      cat_thresh = [ ==0.1 ];

     field = [{
         name = "PROB";
         level = "Z2";
         prob = { name = "TMP"; thresh_lo = 258.15; thresh_hi = 268.15;} }
     ];
}

*I'm not sure if I set the thresh_hi and thresh_lo you are wanting to use as I just picked one that was in the file.   But, the above values pull record number 21.  If I instead set prob = { name = "TMP"; thresh_hi = 258.15;}, it will pull record number 28.

I was able to get it to read data making the changes above.  Let me know if this works for you.  More information about probability and grib can be found [here](https://met.readthedocs.io/en/latest/Users_Guide/config_options.html#fcst), and  we have a grib probability example on [github](https://github.com/dtcenter/MET/blob/main_v11.0/internal/test_unit/config/GridStatConfig_climo_prob).

Tina

**Reply by lindsayrblank** [reply]
Thank you so much! This fixed everything.

**Reply by CPKalb** [reply]
Great!  I'm going to lock the discussion.  It looks like it's already marked as answered!

Tina
